subworkflows/local/happy_benchmark/main.nf

```diff
@@ -0,0 +1,162 @@
+    stratification_tsv // reference channel [val(meta), tsv]
+
+    main:
+
+    versions        = Channel.empty()
+    tagged_variants = Channel.empty()
+
+    input_ch
+        .map{ meta, vcf, _tbi, _truth_vcf, _truth_tbi, _regionsbed, _targets_bed  ->
+            [ meta, vcf ]
+        }
+        .set { test_ch }
+
+    input_ch
+        .map{ meta, _vcf, _tbi, truth_vcf, _truth_tbi, _regionsbed, _targets_bed  ->
+            [ meta, truth_vcf, _regionsbed, _targets_bed ]
+        }
+        .set { truth_ch }
+
+    if (params.preprocess.contains("prepy")){
+
+        // apply prepy if required
+        HAPPY_PREPY(
+            input_ch.map{ meta, vcf, _tbi, _truth_vcf, _truth_tbi, _regionsbed, _targets_bed  ->
+                [ meta, vcf, _regionsbed ]
+            },
+            fasta,
+            fai
+        )
+        versions = versions.mix(HAPPY_PREPY.out.versions.first())
+
+        test_ch = HAPPY_PREPY.out.preprocessed_vcf
+    }
+
+    // apply happy method for benchmarking
+    HAPPY_HAPPY(
+        test_ch.join(truth_ch, failOnDuplicate:true, failOnMismatch:true),
+        fasta,
+        fai,
+        falsepositive_bed,
+        stratification_tsv,
+        stratification_bed
+    )
+    versions = versions.mix(HAPPY_HAPPY.out.versions.first())
+
+    // tag meta and collect summary reports
+    HAPPY_HAPPY.out.summary_csv
+        .map { _meta, file -> tuple([vartype: params.variant_type] + [benchmark_tool: "happy"], file) }
+        .groupTuple()
+        .set{ summary_reports }
+
+    // Subsample TRUTH column from happy results
+    BCFTOOLS_VIEW_TRUTH(
+        HAPPY_HAPPY.out.vcf.join(HAPPY_HAPPY.out.tbi),
+        [],
+        [],
+        []
+        )
+    versions = versions.mix(BCFTOOLS_VIEW_TRUTH.out.versions.first())
+
+    // reheader benchmarking results properly and tag meta
+    BCFTOOLS_REHEADER_1(
+        BCFTOOLS_VIEW_TRUTH.out.vcf.map{ meta, vcf ->
+        [ meta, vcf, [], [] ]
+        },
+        fai
+    )
+    versions = versions.mix(BCFTOOLS_REHEADER_1.out.versions)
+
+    // Subsample QUERY column from happy results
+    BCFTOOLS_VIEW_QUERY(
+        HAPPY_HAPPY.out.vcf.join(HAPPY_HAPPY.out.tbi),
+        [],
+        [],
+        []
+    )
+    versions = versions.mix(BCFTOOLS_VIEW_QUERY.out.versions)
+
+    // reheader benchmarking results properly and tag meta
+    BCFTOOLS_REHEADER_2(
+        BCFTOOLS_VIEW_QUERY.out.vcf.map{ meta, vcf ->
+        [ meta, vcf, [], [] ]
+        },
+        fai
+    )
+    versions = versions.mix(BCFTOOLS_REHEADER_2.out.versions)
+
+    BCFTOOLS_FILTER_TRUTH_TP(
+        BCFTOOLS_REHEADER_1.out.vcf.join(BCFTOOLS_REHEADER_1.out.index)
+    )
+    versions = versions.mix(BCFTOOLS_FILTER_TRUTH_TP.out.versions)
+
+    BCFTOOLS_FILTER_TRUTH_TP.out.vcf
+        .join(BCFTOOLS_FILTER_TRUTH_TP.out.tbi)
+        .map { _meta, file, index -> tuple([vartype: params.variant_type] + [tag: "TP_comp"] + [id: "happy"], file, index) }
```

Comment:
Just my taste. I think you can rewrite this into:
`tuple([vartype: params.variant_type, tag: "TP_comp", id: "happy"], file, index)`